Autodocs

tmccombs opened the pull request:
Add github actions workflow to automate api doc generation.

tmccombs committed:
Add build job to update scaladocs

SrodriguezO commented:
I thought the `release` and `publish` steps of CI only ran on tag pushes; how come they ran here? Also, are the failures there expected?

tmccombs commented:
it's weird that they showed up here. I temporarily added a tag to test the doc publishing, but I think the tag may have been a duplicate of a previous tag, so publishing to sonatype failed. It was a while ago, so I may be misremembering.

SrodriguezO commented on .github/workflows/ci.yml:
```diff
@@ -80,3 +80,51 @@ jobs:
+        with:
+          ref: refs/heads/gh-pages
+          path: pages
+          fetch-depth: 0
```
why do you need to fetch the full history of the pages?

tmccombs replied:
I.... don't remember.

SrodriguezO commented on .github/workflows/ci.yml:
```diff
@@ -80,3 +80,51 @@ jobs:
+          key: ${{ runner.os }}-sbt-${{ hashFiles('**/*.sbt') }}-${{ hashFiles('**/*.scala') }}
+          restore-keys: |
+            ${{ runner.os }}-sbt-${{ hashFiles('**/*.sbt') }}-
+            ${{ runner.os }}-sbt-
```
I feel like I'm missing something with how restore-keys work; from the docs, it sounds like they'll provide a hit if there's any entries in the cache that match this prefix. But that seems like it'd give you false hits? If the main key was missed, that means the Scala source changed. Am I missing something?

TBH this is probably not built enough for a complex cache to be worthwhile. I'd remove the restore-keys form the cache. And in retrospect, I'd remove them from the other jobs' caches too unless you're confident they won't ever cause false hits (and if so, please explain this to me :smile:)

tmccombs replied:
The idea here is that even if the source files are different we can still use the cache for upstream packages.  

You're right though, this won't be built very often. I can remove it.

SrodriguezO reviewed (changes requested):
LGTM; the deep fetch is mostly a nit, the restore keys I'd like to understand before approving or we can remove them (I don't think they add much)

tmccombs committed:
Remove fetch-depth for gh-pages

Because I don't think we really need the full history.

tmccombs committed:
Remove restore-keys for publish-docs

Because it is built infrequently enough that the extra keys probably
aren't worth it.
